armi/bookkeeping/visualization/dumper.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2020 TerraPower, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Abstract base class for visualization file dumpers.
+"""
+from abc import ABC, abstractmethod
+
+from armi.reactor import reactors
+
+class VisFileDumper(ABC):
+    @abstractmethod
+    def dumpState(self, r: reactors.Reactor):
+        """Dump a single reactor state to the vis file."""
+
+    @abstractmethod
+    def initialize(self):
```

Comment:
Since called by the context manager should this be `_initialize`? Or do you expect clients to sometimes call it directly? If so then the `__enter__` thing in the docstring may be confusing

Reply:
Something just feels weird to me about requiring context manageryness from an ABC. Maybe its not that crazy. I think ill just get rid of initialize and finalize, and implementers will just have to define __enter__ and __exit__ 🤷‍♀️